src/many/src/types/ledger.rs

```diff
@@ -220,20 +221,48 @@ impl<'de> Deserialize<'de> for TokenAmount {
                 formatter.write_str("amount in number or string")
             }
 
+            fn visit_i64<E>(self, v: i64) -> Result<Self::Value, E>
```

Comment:
Do we have a use-case for this one? `i*` are not defined for `TokenAmount`.

Reply:
Yes, one of the serde crate I want to use uses `i64` for numbers. I should implement the others for consistency.